app/src/lib/stores/app-store.ts

```diff
@@ -1961,15 +1980,19 @@ export class AppStore extends TypedBaseStore<IAppState> {
     }
 
     const gitStore = this.getGitStore(repository)
-
     const status = await gitStore.loadStatus()
     if (status === null) {
       lookup.delete(repository.id)
       return
     }
 
     this.withAuthenticatingUser(repository, async (repo, account) => {
-      if (this.shouldBackgroundFetch(repo)) {
+      const lastPush = await this.inferLastPushForRepository(
+        this.accounts,
+        gitStore,
+        repository
+      )
+      if (this.shouldBackgroundFetch(repo, lastPush)) {
```

Comment:
There's some extra work that we need to do in here for these dormant repositories:

 - `gitStore.updateLastFetched()` - this pokes at the `.git/FETCH_HEAD` file on disk
 - `gitStore.loadRemotes()` - this hydrates the remotes information from the repository

If we skip the first item, the `lastFetched` value is always empty when we need to compare it inside `this.shouldBackgroundFetch`. If we skip the second, the inner `gitStore.fetch` is no-op because it doesn't have any remotes to work with.

I think we should move `this.shouldBackgroundFetch` out of `this.withAuthenticatingUser` - remember the large call stack from https://github.com/desktop/desktop/pull/5297#issue-205091781 - because we only need to do this work when it's going to fetch. Here's a diff of how I think this would look:

```diff
diff --git a/app/src/lib/stores/app-store.ts b/app/src/lib/stores/app-store.ts
index a739b0e48..4ca552eb2 100644
--- a/app/src/lib/stores/app-store.ts
+++ b/app/src/lib/stores/app-store.ts
@@ -1986,18 +1986,23 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return
     }
 
-    this.withAuthenticatingUser(repository, async (repo, account) => {
-      const lastPush = await this.inferLastPushForRepository(
-        this.accounts,
-        gitStore,
-        repository
-      )
-      if (this.shouldBackgroundFetch(repo, lastPush)) {
-        await gitStore.performFailableOperation(() => {
+    const lastPush = await this.inferLastPushForRepository(
+      this.accounts,
+      gitStore,
+      repository
+    )
+
+    await gitStore.updateLastFetched()
+
+    if (this.shouldBackgroundFetch(repository, lastPush)) {
+      await gitStore.loadRemotes()
+
+      await this.withAuthenticatingUser(repository, async (repo, account) => {
+        return gitStore.performFailableOperation(() => {
           return gitStore.fetch(account, true)
         })
-      }
-    })
+      })
+    }
 
     lookup.set(repository.id, {
       aheadBehind: gitStore.aheadBehind,
```

Also ensure you `await this.withAuthenticatingUser` because that ensures we keep this running sequentially.